File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/interfaces/interfaceClassWithInterfaceFieldsOfDifferentType/expected/types/Vegetarian.java
package com.netflix.graphql.dgs.codegen.java.testcases.interfaces.interfaceClassWithInterfaceFieldsOfDifferentType.expected.types;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.List;
import java.util.Objects;

@JsonTypeInfo(
    use = JsonTypeInfo.Id.NONE
)
public class Vegetarian implements com.netflix.graphql.dgs.codegen.java.testcases.interfaces.interfaceClassWithInterfaceFieldsOfDifferentType.expected.types.Diet {
  private String calories;

  private List<String> vegetables;

  public Vegetarian() {
  }

  public Vegetarian(String calories, List<String> vegetables) {
    this.calories = calories;
    this.vegetables = vegetables;
  }

  public String getCalories() {
    return calories;
  }

  public void setCalories(String calories) {
    this.calories = calories;
  }

  public List<String> getVegetables() {
    return vegetables;
  }

  public void setVegetables(List<String> vegetables) {
    this.vegetables = vegetables;
  }

  @Override
  public String toString() {
    return "Vegetarian{calories='" + calories + "', vegetables='" + vegetables + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Vegetarian that = (Vegetarian) o;
    return Objects.equals(calories, that.calories) &&
        Objects.equals(vegetables, that.vegetables);
  }

  @Override
  public int hashCode() {
    return Objects.hash(calories, vegetables);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private String calories;

    private List<String> vegetables;

    public Vegetarian build() {
      Vegetarian result = new Vegetarian();
      result.calories = this.calories;
      result.vegetables = this.vegetables;
      return result;
    }

    public Builder calories(String calories) {
      this.calories = calories;
      return this;
    }

    public Builder vegetables(List<String> vegetables) {
      this.vegetables = vegetables;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/interfaceClassWithInterfaceFieldsOfDifferentType/expected/types/Vegetarian.kt
package com.netflix.graphql.dgs.codegen.cases.interfaceClassWithInterfaceFieldsOfDifferentType.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import java.lang.IllegalStateException
import kotlin.String
import kotlin.Suppress
import kotlin.collections.List
import kotlin.jvm.JvmName
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize as FasterxmlJacksonDatabindAnnotationJsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder as FasterxmlJacksonDatabindAnnotationJsonPOJOBuilder
import tools.jackson.databind.`annotation`.JsonDeserialize as ToolsJacksonDatabindAnnotationJsonDeserialize
import tools.jackson.databind.`annotation`.JsonPOJOBuilder as ToolsJacksonDatabindAnnotationJsonPOJOBuilder

@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@FasterxmlJacksonDatabindAnnotationJsonDeserialize(builder = Vegetarian.Builder::class)
@ToolsJacksonDatabindAnnotationJsonDeserialize(builder = Vegetarian.Builder::class)
public class Vegetarian(
  calories: () -> String? = caloriesDefault,
  vegetables: () -> List<String?>? = vegetablesDefault,
) : Diet {
  private val __calories: () -> String? = calories

  private val __vegetables: () -> List<String?>? = vegetables

  @Suppress("INAPPLICABLE_JVM_NAME")
  @get:JvmName("getCalories")
  override val calories: String?
    get() = __calories.invoke()

  @get:JvmName("getVegetables")
  public val vegetables: List<String?>?
    get() = __vegetables.invoke()

  public companion object {
    private val caloriesDefault: () -> String? = 
        { throw IllegalStateException("Field `calories` was not requested") }

    private val vegetablesDefault: () -> List<String?>? = 
        { throw IllegalStateException("Field `vegetables` was not requested") }
  }

  @FasterxmlJacksonDatabindAnnotationJsonPOJOBuilder
  @ToolsJacksonDatabindAnnotationJsonPOJOBuilder
  @JsonIgnoreProperties("__typename")
  public class Builder {
    private var calories: () -> String? = caloriesDefault

    private var vegetables: () -> List<String?>? = vegetablesDefault

    @JsonProperty("calories")
    public fun withCalories(calories: String?): Builder = this.apply {
      this.calories = { calories }
    }

    @JsonProperty("vegetables")
    public fun withVegetables(vegetables: List<String?>?): Builder = this.apply {
      this.vegetables = { vegetables }
    }

    public fun build(): Vegetarian = Vegetarian(
      calories = calories,
      vegetables = vegetables,
    )
  }
}
